All things Command Processing

kontacky01 opened the pull request:
Update
=>i added all things commandprocessing draft1(not finished yet)
=>Changes i made to other classes
     1. changed the class Transition in gameengine
     2. i added a setState method to class player to be able to move a player to different states
     3. i changed the wording of commans to match asg2 pdf command wording ex: quit instead of end, gamestart inlieu of assigncountries
     4. i removed the win transition to go to winstate because this can only be achieved through playing the game that is if someone has defeated others not just by a user entering a command

kontacky01 committed:
Update
i modified gameengine in particular class transition to accomodate some asg2 functionalities. now a transition object has 3 attributes instead of 2. currentState, command, and nextState

kontacky01 committed:
Update
=>i added all things commandprocessing draft1(not finished yet)
=>Changes i made to other classes
     1. changed the class Transition in gameengine
     2. i added a setState method to class player to be able to move a player to different states
     3. i changed the wording of commans to match asg2 pdf command wording ex: quit instead of end, gamestart inlieu of assigncountries
     4. i removed the win transition to go to winstate because this can only be achieved through playing the game that is if someone has defeated others not just by a user entering a command

LaraMo commented on src/CommandProcessing/CommandProcessing.cpp:
```diff
@@ -0,0 +1,197 @@
+ */
+    Command CommandProcessor::getCommandForOtherObjects(){
+        if (!commands.empty()) {
+            //TODO
```
what should be done here?:)

kontacky01 replied:
oh i just wrote to do as a self note because am not yet done with the code. i just wanted to publish what i have to give everyone an idea of what am working on in case they need it for their work. i'll commit again once i finish

LaraMo commented on src/CommandProcessing/CommandProcessing.cpp:
```diff
@@ -0,0 +1,197 @@
+    void FileCommandProcessorAdapter::validate(Command* cmd, Player* p){
+        bool isValid = false;
+        vector<Transition*> transitions = initializeGameTransitions();
+        for (int i = 0; i < transitions.size(); i++) {
```
very similar to     `void CommandProcessor::validate(Command* cmd, Player* p)` , could this be a helper method instead?

kontacky01 replied:
i thought about it too. i might make validate() and getCommandForOtherObjects() heleper methods for both FileCommandProcessorAdapter and CommandProcessor classes since they share them

LaraMo reviewed (commented):
great job Natasha!!!!!

kontacky01 committed:
Update
what i have so far...still coding not done yet. so i still need to do some debugging..plus i can't run stuff on my end so i do blind coding most part...maindriver errors and orderlist
...will proceed later today

kontacky01 committed:
Update
current code that i have atm!
